spec/facter/resolvers/windows/networking_spec.rb

```diff
@@ -217,6 +218,15 @@
       end
       let(:domain) { 'my_domain' }
 
+      def stub_utf16le_bytes(ptr, str)
+        i = 0
+        str.encode(Encoding::UTF_16LE).each_char do |ch|
+          allow(ptr).to receive(:get_bytes).with(i, 2).and_return(ch)
+          i += 2
+        end
+        allow(ptr).to receive(:get_bytes).with(i, 2).and_return("\0\0")
```

Comment:
The line I added isn't quite right. `"\0\0"` is a UTF-8 encoded string representing two null **characters**. We really should be returning a UTF-16LE string for a single null wide character like:

```suggestion
        allow(ptr).to receive(:get_bytes).with(i, 2).and_return("\0".encode(Encoding::UTF_16LE))
```

It just so happens that they have the same byte sequence:

```
irb(main):001:0> "\0".encode(Encoding::UTF_16LE).bytes == "\0\0".bytes
=> true
```

Or even better, pull the null wide character into a constant or let, like we do in puppet:
https://github.com/puppetlabs/puppet/blob/2f14ddd269c26cba36771d9b3464c2c29422e46a/lib/puppet/ffi/windows/api_types.rb#L23